eval/eval.go

```diff
@@ -77,8 +77,9 @@ func EvalEnvironment(
 	decrypter Decrypter,
 	providers ProviderLoader,
 	environments EnvironmentLoader,
+	context esc.Value,
```

Comment:
how about we make this a `map[string]esc.Value`? we could always relax it later if we need to, but I think that'll simplify callers of this function a bit and it's sufficient for our needs.

Reply:
Done, I end up wrapping the `map[string]esc.Value` into a `Value` internally to keep it simple (otherwise I would have to write a new `unexport` to handle that case and/or change how `evaluateExprAccess` process the value. I believe this makes it more straightforward.